language/proto/generate.go

```diff
@@ -41,13 +41,29 @@ func (_ *protoLang) GenerateRules(args language.GenerateArgs) language.GenerateR
 			regularProtoFiles = append(regularProtoFiles, name)
 		}
 	}
-	var genProtoFiles []string
+
+	// Some of the generated files may have been consumed by other rules
+	consumedFileSet := make(map[string]bool)
+	for _, r := range args.OtherGen {
```

Comment:
@jayconrod I think this can be more specific by having:
```Go
if r.Kind() != "proto_library" {
  continue
}
```
It should be an invalid use case for two proto_library rules to compile the same proto file. Gazelle won't be able to resolve that. It would still allow filegroup or pkg_tar rules to consume the proto files

Reply:
Hmm, okay, I think I agree.